msbuild/Xamarin.MacDev.Tasks.Core/Tasks/MergeAppBundlesTaskBase.cs

```diff
@@ -282,6 +282,11 @@ bool MergeAppBundles ()
 					BundlePath = fullInput,
 					SpecificSubdirectory = specificSubdirectory,
 				};
+				// Remove any files inside directories which are symlinks (we only need to process the symlink itself)
+				var symlinkDirectories = files.Where (v => PathUtils.IsSymlink (v) && Directory.Exists (v));
+				if (symlinkDirectories.Any ()) {
+					files = files.Where (file => !symlinkDirectories.Any (dir => file.StartsWith (dir + Path.DirectorySeparatorChar))).ToArray ();
```

Comment:
Wouldn't it be better to have a hashset with a custom comparer here (avoids allocations of all the strings and closures)? In example, are these tasks run on Unix only?

Reply:
Yes, this algorithm isn't the fastest. That said, it's not a very huge dataset either (usually a few hundred files at most), so I'm not sure how much of a difference a faster (and more complex) algorithm will be, and if it'll be worth the time spent implementing it.

Reply:
Those hundreds of files across many builds called all over CI bots and stuff do add up :D